WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -220,7 +220,32 @@
     <% end %>
 
     <%= f.input :currency_code, collection: Money::Currency.table.values, label_method: lambda { |c| c[:name]  }, value_method: lambda { |c| c[:iso_code] }  %>
-    <%= f.input :base_entry_fee_lowest_denomination %>
+    <%= f.input :base_entry_fee_lowest_denomination, input_html: { type: "text" } %>
+    <script>
+
+      // Setup the currency mask on the fee field
+      $(function() {
+        wca.applyCurrencyMask('init',
+                              '#competition_base_entry_fee_lowest_denomination',
+                              $('#competition_currency_code').val());
+      })
+
+      // Setup form onsubmit callback
+      $('#edit-competition-form').submit(function() {
```

Comment:
I did that, this + the custom input field were really great suggestions!
It also get rid of one issue I noticed: when directly applying the mask on the actual field send to the server, it would systematically mark the form as modified, therefore triggering the js alert "are you sure you want to leave blah blah, the form has been modified" on F5. Here the field with the mask is not watched, so this alert only occurs when a change actually has been made!